daq/faucet_states_collector.py

```diff
@@ -108,6 +111,82 @@ def get_switch(self, switch_name):
 
         return switch_map
 
+    def get_active_host_route(self, src_mac, dst_mac):
+        """give two MAC addresses in the core network, find the active route between them"""
+        def get_switches_ports_from_link(link_map):
+            return (link_map["port_map"]["dp_a"], int(link_map["port_map"]["port_a"][5:])), \
+                   (link_map["port_map"]["dp_z"], int(link_map["port_map"]["port_z"][5:]))
+
+        def check_and_insert_link(sw_1, port_1, sw_2, port_2):
+            if src_learned_switches.get(sw_1, {}).get(KEY_MAC_LEARNING_PORT, "") == port_1 and \
+                    dst_learned_switches.get(sw_2, {}).get(KEY_MAC_LEARNING_PORT, "") == port_2:
+                next_hops[sw_2] = sw_1
+
+        def get_graph():
+            for link_map in self.topo_state.get(TOPOLOGY_GRAPH).get("links", []):
+                if not link_map:
+                    continue
+                sw1_p1, sw2_p2 = get_switches_ports_from_link(link_map)
+                check_and_insert_link(*sw1_p1, *sw2_p2)
+                check_and_insert_link(*sw2_p2, *sw1_p1)
+
+        def get_access_switches():
+            src_switches_ports = {}
+            dst_switches_ports = {}
+
+            for switch, port_map in src_learned_switches.items():
+                src_switches_ports[switch] = port_map[KEY_MAC_LEARNING_PORT]
+
+            for switch, port_map in dst_learned_switches.items():
+                dst_switches_ports[switch] = port_map[KEY_MAC_LEARNING_PORT]
+
+            for link_map in self.topo_state.get(TOPOLOGY_GRAPH).get("links", []):
+                if not link_map:
+                    continue
+                (sw_1, port_1), (sw_2, port_2) = get_switches_ports_from_link(link_map)
+                if src_switches_ports.get(sw_1, "") == port_1:
+                    src_switches_ports.pop(sw_1)
+                if src_switches_ports.get(sw_2, "") == port_2:
+                    src_switches_ports.pop(sw_2)
+                if dst_switches_ports.get(sw_1, "") == port_1:
+                    dst_switches_ports.pop(sw_1)
+                if dst_switches_ports.get(sw_2, "") == port_2:
+                    dst_switches_ports.pop(sw_2)
+
+            return src_switches_ports.popitem(), dst_switches_ports.popitem()
+
+        res = {"Path", []}
+        next_hops = {}
+
+        if src_mac not in self.system_states.get(KEY_LEARNED_MACS, {}) or \
+                dst_mac not in self.system_states.get(KEY_LEARNED_MACS, {}):
+            return res
+
+        src_learned_switches = \
```

Comment:
find a way to remove line breaks (local variable)

Reply:
Done